src/Language/Printer.php

```diff
@@ -86,379 +92,356 @@ public static function doPrint($ast)
-                            ],
-                            ' '
-                        );
-                    },
+    /**
+     * @return array<Node>
+     */
+    protected function listToArray(?NodeList $list) : array
+    {
+        return isset($list) ? iterator_to_array($list) : [];
+    }
 
-                    NodeKind::ENUM_TYPE_EXTENSION => function (EnumTypeExtensionNode $def) {
-                        return $this->join(
-                            [
-                                'extend enum',
-                                $def->name,
-                                $this->join($def->directives, ' '),
-                                $this->block($def->values),
-                            ],
-                            ' '
-                        );
-                    },
+    protected function printList(?NodeList $list, $separator = '') : string
+    {
+        return $this->printArray($this->listToArray($list), $separator);
+    }
 
-                    NodeKind::INPUT_OBJECT_TYPE_EXTENSION => function (InputObjectTypeExtensionNode $def) {
-                        return $this->join(
-                            [
-                                'extend input',
-                                $def->name,
-                                $this->join($def->directives, ' '),
-                                $this->block($def->fields),
-                            ],
-                            ' '
-                        );
-                    },
+    protected function printListBlock(?NodeList $list) : string
+    {
+        return $this->block(array_map(function ($item) : string {
+            return $this->p($item);
+        }, $this->listToArray($list)));
+    }
 
-                    NodeKind::DIRECTIVE_DEFINITION => $this->addDescription(function (DirectiveDefinitionNode $def) {
-                        $noIndent = Utils::every($def->arguments, static function (string $arg) : bool {
-                            return strpos($arg, "\n") === false;
-                        });
+    /**
+     * @param array<Node> $list
+     */
+    protected function printArray(array $list, string $separator = '') : string
+    {
+        return $this->join(array_map(function ($item) : string {
+            return $this->p($item);
+        }, $list), $separator);
+    }
 
-                        return 'directive @'
-                            . $def->name
-                            . ($noIndent
-                                ? $this->wrap('(', $this->join($def->arguments, ', '), ')')
-                                : $this->wrap("(\n", $this->indent($this->join($def->arguments, "\n")), "\n"))
-                            . ($def->repeatable ? ' repeatable' : '')
-                            . ' on ' . $this->join($def->locations, ' | ');
-                    }),
-                ],
-            ]
-        );
+    public function p(?Node $node, bool $isDescription = false) : string
```

Comment:
```suggestion
    public function print(?Node $node, bool $isDescription = false) : string
```

Reply:
Or `printNode`

Reply:
No problem if you really want it, but I chose `p` because I now use it inline all over the place, and it helps keep the code a bit more concise. Think it over a bit more and let me know what you think.